app/controllers/courses_controller.rb

```diff
@@ -127,4 +132,13 @@ def save_tag
       session[:applicant_tag] = params[:tag]
     end
   end
+
+  # This method is to handle when a coach visits a course whey they are not a coach from a courses/cohorts where they are coach
+  def handle_coach_routing_error
```

Comment:
Let's remove this and have the presenter handle the available links to the coach in the course pages